### ISSUE TITLE
No more location updates after resume

### ISSUE THREAD

**Issue by grote** [mapbox/mapbox-plugins-android#371]
I am seeing an issue with the Location Plugin 0.4 as used by the Mapbox Demo app. (So maybe this ticket belongs to the app and not the library, I am not sure). The problem is that the `AndroidLocationEngine` is not getting any location updates after the activity was stopped and started again.

Then starting the activity for the first time, things are as they should:

```
V/AndroidLocationEngine: Initializing.
D/AndroidLocationEngine: Priority set to 0 (current provider is passive).
V/AndroidLocationEngine: Activating.
D/AndroidLocationEngine: Priority set to 3 (current provider is gps).
V/AndroidLocationEngine: Activating.
V/AndroidLocationEngine: New location received.
```

Location updates are received regularly after this. However, if the app is put in the background and then started again. The same activity is re-used and it stops receiving location updates.

When rotating the screen, the activity is recreated and it starts receiving new location updates again.

**Comment by grote** [comment]
It looks like the demo app calls `locationPlugin.onStart()` in [`onStart()`](https://github.com/mapbox/mapbox-android-demo/blob/master/MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/plugins/LocationPluginActivity.java#L133), but does not explicitly request location updates.

This only seems to happen in `onConnected()` which isn't called when the activity is re-started.

Explicitly requesting location updates in `onStart()` seems to solve the issue:
```java
    if (locationEngine != null) {
      locationEngine.requestLocationUpdates();
    }
```

**Comment by cammace** [comment]
Will try taking a look into this issue today. Your initial thought sounds to be the correct issue. A workaround might be to hold an instance of the same locationEngine locally and request location updates yourself (of course until this gets fixed).

**Comment by cammace** [comment]
We just released 0.5.0-beta.1 with a ton of changes, mainly the update to the Map SDK 6.0. While it will require work to make the upgrade, I believe the fix to your issue is found in #307. Can you try this out and confirm this fixes the issue?

**Comment by cammace** [comment]
Tested on my end and haven't run into this issue anymore. Closing
